### ISSUE TITLE
DirectoryName::addr_attr generates uncommon name structure

### ISSUE THREAD

**Issue by Albibek** [Devolutions/picky-rs#44]
Hello.

I was trying to use picky as a library and found a behaviour that is not very clear in RFC, but does not conform to common practice of creating certificate subjects.

The DirectoryName and Name structures has only 1 function for adding more subject components - `add_attr`, I would expect it to make a subject like in all other certificates: a sequence of sets of name-value pairs(which are also sequences). But it currently generates only one set, adding all new pairs to the sequence inside that same set, resulting into sequence with on set containtng multiple sequences.

Here is the code example to reproduce the issue (partially copy-pasted from `picky/src/x509/certificate.rs`):
```rust
    #[test]
    fn correct_name_handling() {
        let root_key = parse_key(crate::test_files::RSA_2048_PK_1);

        let mut name = DirectoryName::new_common_name("TheFuture.usodakedo Root CA");
        name.add_attr(crate::x509::name::NameAttr::LocalityName, "SouthPole");
        name.add_attr(crate::x509::name::NameAttr::CountryName, "WO");
        let root = CertificateBuilder::new()
            .valididy(UTCDate::ymd(2065, 6, 15).unwrap(), UTCDate::ymd(2070, 6, 15).unwrap())
            .self_signed(name, &root_key)
            .ca(true)
            .signature_hash_type(SignatureAlgorithm::RsaPkcs1v15(HashAlgorithm::SHA2_512))
            .key_id_gen_method(KeyIdGenMethod::SPKFullDER(HashAlgorithm::SHA2_384))
            .build()
            .expect("couldn't build root ca");
        assert_eq!(root.ty(), CertType::Root);
        use std::io::Write;
        let mut file = std::fs::File::create("root.pem").unwrap();
        file.write_all(crate::pem::to_pem("CERTIFICATE", &root.to_der().unwrap()).as_bytes())
            .unwrap();
        file.flush().unwrap();
    }
```

Here's the results of that test code:
```
# Note the plus sign in openssl representation, in certificates it's usually comma, not plus
$ openssl x509 -subject -noout  < root.pem
subject=CN = TheFuture.usodakedo Root CA + L = SouthPole + C = WO
```
The subject field looks like this in asn1:
```
$ openssl asn1parse -i -in root.pem
...
   34:d=2  hl=2 l=  67 cons:   SEQUENCE
   36:d=3  hl=2 l=  65 cons:    SET
   38:d=4  hl=2 l=  34 cons:     SEQUENCE
   40:d=5  hl=2 l=   3 prim:      OBJECT            :commonName
   45:d=5  hl=2 l=  27 prim:      UTF8STRING        :TheFuture.usodakedo Root CA
   74:d=4  hl=2 l=  16 cons:     SEQUENCE
   76:d=5  hl=2 l=   3 prim:      OBJECT            :localityName
   81:d=5  hl=2 l=   9 prim:      UTF8STRING        :SouthPole
   92:d=4  hl=2 l=   9 cons:     SEQUENCE
   94:d=5  hl=2 l=   3 prim:      OBJECT            :countryName
   99:d=5  hl=2 l=   2 prim:      UTF8STRING        :WO
...
```

Here's a standard CA certificate subject:
```
$ openssl asn1parse -i < /etc/ssl/certs/DigiCert_Assured_ID_Root_CA.pem
  181:d=2  hl=2 l= 101 cons:   SEQUENCE
  183:d=3  hl=2 l=  11 cons:    SET
  185:d=4  hl=2 l=   9 cons:     SEQUENCE
  187:d=5  hl=2 l=   3 prim:      OBJECT            :countryName
  192:d=5  hl=2 l=   2 prim:      PRINTABLESTRING   :US
  196:d=3  hl=2 l=  21 cons:    SET
  198:d=4  hl=2 l=  19 cons:     SEQUENCE
  200:d=5  hl=2 l=   3 prim:      OBJECT            :organizationName
  205:d=5  hl=2 l=  12 prim:      PRINTABLESTRING   :DigiCert Inc
  219:d=3  hl=2 l=  25 cons:    SET
  221:d=4  hl=2 l=  23 cons:     SEQUENCE
  223:d=5  hl=2 l=   3 prim:      OBJECT            :organizationalUnitName
  228:d=5  hl=2 l=  16 prim:      PRINTABLESTRING   :www.digicert.com
  246:d=3  hl=2 l=  36 cons:    SET
  248:d=4  hl=2 l=  34 cons:     SEQUENCE
  250:d=5  hl=2 l=   3 prim:      OBJECT            :commonName
  255:d=5  hl=2 l=  27 prim:      PRINTABLESTRING   :DigiCert Assured ID Root CA
```

How to fix:
If this behaviour was not intended for some reason, I think this line:
https://github.com/Devolutions/picky-rs/blob/master/picky-asn1-x509/src/name.rs#L97

```rust
   ((self.0).0)[0].0.push(ty_val);
```

should be replaced with something like this:
```rust
   let set_val = Asn1SetOf(vec![ty_val]);
   ((self.0).0).push(set_val);
```

**Comment by awakecoding** [comment]
@bcortier-devolutions if you can take a look

**Comment by bcortier-devolutions** [comment]
Hi @Albibek ,
You're entirely right about this, we should conform to common practices, especially if this doesn't contradict with RFC.
Your fix looks good to me!

**Comment by bcortier-devolutions** [comment]
Addressed by #46
